app/include/zmk/display.h

```diff
@@ -6,4 +6,48 @@
 
 #pragma once
 
-int zmk_display_init();
\ No newline at end of file
+struct k_work_q *zmk_display_work_q();
+
+bool zmk_display_is_initialized();
+int zmk_display_init();
+
+/**
+ * @brief Macro to define a ZMK event listener that handles the thread safety of fetching
+ * the necessary state from the system work queue context, invoking a work callback
+ * in the display queue context, and properly accessing that state safely when performing
+ * display/LVGL updates.
+ * @param listener THe ZMK Event manager listener name
+ * @param state_type The struct/enum type used to store/transfer state.
+ * @param cb The callback to invoke in the dispaly queue context to update the UI
+ * @param state_func The callback function to invoke to fetch the updated state from ZMK core
+ * @returns Generates a function listener##_init that should be called by the widget once ready to
```

Comment:
Seems a bit weird to list this in `@returns` since it doesn't behave like a function and doesn't "return" the _init() function.

Reply:
Yeah, I didn't find anything authoritative on how to document functions generated by a macro, and fell back to this. Is there something here I missed?

Reply:
Switched this to at least use `@retval` which let's you *name* the returned value you're documenting, which seems better. Thoughts?